### PR TITLE
Project Phases Table

### PR DESCRIPTION
**Pull request by mateoclarke**
addressed part of https://github.com/cityofaustin/atd-data-tech/issues/4332

This PR add as "Project Phases" table to the Timeline Tab of the Project Details Page. The feature uses [Material Table](https://material-table.com/#/docs/features/editable) as a dependency because of the out of the box UI available for creating, editing, and deleting table items.

Note: The delete feature will not work in the staging instance  until the permissions/Hasura config settings are applied.

### Add
Screen Shot 2021-01-29 at 2 19 07 AM

### Edit
Screen Shot 2021-01-29 at 2 18 46 AM

### Delete
Screen Shot 2021-01-29 at 2 17 01 AM

### CODE REVIEW

**Comment by amenity** [comment]
So slick, @mateoclarke! ✨ 😎

- What do you think about moving the "+" from the upper right to below the table as a [FAB](https://material.io/components/buttons-floating-action-button)? 

~I'm not able to delete any rows a~

**Review by amenity — APPROVED** [review]

File: moped-editor/src/views/projects/projectView/ProjectTimelineViewQueryConf.js
Comment (sergiogcx) [inline moped-editor/src/views/projects/projectView/ProjectTimelineViewQueryConf.js:89]: What is the purpose of this file? It looks like it was meant for a timeline view, but at the same time I don't see it being used or imported in any other files?

Comment (sergiogcx) [inline moped-editor/src/views/projects/projectView/ProjectTimelineViewQueryConf.js:89]: What is the purpose of this file? It looks like it was meant for a timeline view, but at the same time I don't see it being used or imported in any other files? It also seems to reference the moped_project table and its attributes...

File: moped-database/seeds/1602292389297_initial_seed.sql
Comment (sergiogcx) [inline moped-database/seeds/1602292389297_initial_seed.sql:53]: Thanks for sorting those out! 🙏

**Review by sergiogcx — CHANGES_REQUESTED** [review]
It works flawlessly! 💯 🥇 

Just a question about the queryconf file and happy to approve once resolved

**Review by zappfyllc — APPROVED** [review]
@mateoclarke will echo everyone else and say this looks slick! I like this table better than the default MUI table that Leslie and I placed in the initial Project Teams page. Very cool!

Requests:
- Agree with @amenity regarding the initial placement of the add icon.
- Can you transform the UI values for "Active?" to Yes/No instead of True/False? Fine with true/false in the database but I think generally yes/no is preferable for user-facing copy.
- Default sorting: can you make this sort chronologically from newest to oldest phase instead of by phase name?

Additional Requirements (may be best handled by a separate issue, will defer to you): 
- User cannot add a phase name that's already been added.
- If a user adds an active phase, set any other active phases to inactive.
- Current phase stored in moped_projects should automatically reflect the newest phase chronologically where "active is true"

Comments:
- When I first clicked the add button to add in a phase, I stupidly forgot to click the checkmark to save and lost the row. Totally user error but I'm just flagging it in case we run into that error during user testing.
- The slight lag between entering a new row and saving it, and then the row appearing is something I'm noting. Is this because we're waiting to see if the mutation is successful (otherwise it would display an error)? Really not a big deal but just thought I'd point out if anyone else thought anything of it.

@mateoclarke we can add any of these that are more complex to a subsequent improvements issue so we can go ahead and merge: nothing major sticks out to me that would stop us from getting it in front of users. Just let me know what is left to do and I can add in the issue (or go ahead and do it and we'll discuss in grooming). Awesome job, and my deep thanks for wading through my issues and separating out the phases table so you could ship it without the milestones requirements!

**Comment by johnclary** [comment]
awesome work, @mateoclarke! To raise one more detail—it would be nice if the date-picker calendar opened automatically when a date field is active. if y'all agree, we might move this into a separate issue to address it globally since we need this enhancement in Sergio's advanced table filters and will need it elsewhere.

File: moped-editor/src/views/projects/projectView/ProjectTimelineViewQueryConf.js
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectTimelineViewQueryConf.js:5]: The update to Apollo fixed this issue for the staff and project queries so I took out this comment in those configs. I know that no cache is the behavior that we will want in some places so just giving an update.

**Review by mddilley — APPROVED** [review]
Material Table works great for this feature, and it works great! Awesome to see another project summary tab filled in. 🙌 

I noticed a small regression in Hasura permissions for a couple fields that I mentioned below and need to be patched before merge. They were hidden in the formatting updates to the tables yaml.

File: moped-editor/src/views/projects/projectView/ProjectTimelineViewQueryConf.js
Comment (mateoclarke) [reply moped-editor/src/views/projects/projectView/ProjectTimelineViewQueryConf.js:89]: hmm, you're right. I think I copy/pasted this file before changing directions on which react table library to use. Thanks for catching this.

**Comment by mateoclarke** [comment]
Ok, I've addressed the code feedback from Sergio and Mike and taken inventory of the enhancements/changes requested by Amenity, John and JD. Going to start tackling them in a new quick PR and make a "Project Phases Tables Enhancements" for the ones that are more in depth.

**Pull request merged**
